### ISSUE TITLE
[igxButton]: cannot change button theme with css-vars

### ISSUE THREAD

**Issue by RivaIvanova** [IgniteUI/igniteui-angular#13185]
## Description  
If we want to set a global button theme, using `css-vars` does not apply the custom theme, however, overriding the theme, i.e., `@include button($custom-button-theme);` works.

 * igniteui-angular version: 16.0.x, 15.1.x, 14.2.x
 * browser: any

## Steps to reproduce  

1. Open this [demo](https://stackblitz.com/edit/df2pfl?file=src%2Fapp%2Fnavbar%2Fnavbar.component.html,src%2Fstyles.scss).
2. The buttons use the default theme, however, the `igx-navbar` and `igx-avatar` use the custom themes.

## Result  
![button1](https://github.com/IgniteUI/igniteui-angular/assets/49126110/561137bd-aa2e-4ebc-bba1-106c1ae83df6)

## Expected result  
![button2](https://github.com/IgniteUI/igniteui-angular/assets/49126110/73700d8c-eed6-4f64-aeef-61799f4d1d86)

**Comment by desig9stein** [comment]
@simeonoff I am not sure how to handle this, if I add a `$selector` inside the theme to scope the variables it will affect all button types. This means if the user changes the button background it will affect all buttons.

@RivaIvanova To fix the issue on your side before we figure out how to handle it you have to simply scope your theme to one of the button classes `igx-button--flat`, `igx-button--raised`...  or give your button a custom class.

``` HTML
$button-theme: button-theme(
    $background: red
);

.igx-button--raised {
    @include css-vars($button-theme);
}
```

**Comment by simeonoff** [comment]
@RivaIvanova **This is the expected behavior for the button theme.**

### Long explanation:

First of all, the button theme is a little quirky. Producing a button theme (using the `button-theme` function) is non-deterministic process, overwriting all button variants (flat, outlined, raised, floating, and icon). The `button-theme` function returns a map of properties that can either be consumed by the `button` mixin or the `css-vars` mixin. Unlike the `button` mixin, the `css-vars` mixin only includes the properties as CSS variables scoped to a given parent selector. The `button` mixin, however, will overwrite all previously defined CSS rules for all buttons, hence, correctly targeting all button variants by selector explicitly. You, as the user, have to be conscious about which buttons you want to style with the set of properties produced by the `button-theme` in combination with `css-vars`.

### TLDR:
When using `css-vars`, make sure you scope the produced theme to the only the buttons you want to style to avoid overwriting every other button variant theme unintentionally.

Example:

```html
<div class="raised-buttons">
    <button igxButton="raised">Test</button>
</div>
```

```scss
$raised-button: button-theme(
    $background: red
);

.raised-button {
    @include css-vars($raised-button);
}
```

In this way, only the buttons inside the `.raised-buttons` container will be styled using your custom theme, giving you the opportunity to support different themes for different button variants.

If you want all buttons variants to use the same theme regardless, you can scope it to a parent selector like the `body`, or the `app-root` component.

I hope that helps.

P.S.:

I have an [updated sample](https://stackblitz.com/edit/df2pfl-ax331k?file=src%2Fstyles.scss) to showcase how you can use what's been said above.

**Comment by RivaIvanova** [comment]
@simeonoff Thanks for the detailed explanation and suggestion. 🙂 It seemed a bit odd when comparing it to the other components from the sample, but now it makes more sense to me for this behavior to be expected when it comes to the button theme.
